Mathlib/Analysis/Convex/SpecificFunctions/Basic.lean

```diff
@@ -147,57 +138,117 @@ theorem one_add_mul_self_le_rpow_one_add {s : ℝ} (hs : -1 ≤ s) {p : ℝ} (hp
-    have : 1 - (1 + (x / y - 1)) ^ p < -p * (x / y - 1) := by
-      linarith [one_add_mul_self_lt_rpow_one_add hyx''''' hyx'''.ne hp]
-    rw [div_lt_iff h3, ← div_lt_div_right hy']
-    convert this using 1
-    · have H : (x / y) ^ p = x ^ p / y ^ p := div_rpow hx hy.le _
-      ring_nf at H ⊢
-      field_simp at H ⊢
-      linear_combination H
-    · ring_nf at H1 ⊢
-      field_simp
-      linear_combination p * (-y + x) * H1
-  · have hyz' : 0 < z - y := by linarith only [hyz]
-    have hyz'' : 1 < z / y := by rwa [one_lt_div hy]
-    have hyz''' : 0 < z / y - 1 := by linarith only [hyz'']
-    have hyz'''' : -1 ≤ z / y - 1 := by linarith only [hyz'']
-    have : p * (z / y - 1) < (1 + (z / y - 1)) ^ p - 1 := by
-      linarith [one_add_mul_self_lt_rpow_one_add hyz'''' hyz'''.ne' hp]
-    rw [lt_div_iff hyz', ← div_lt_div_right hy']
-    convert this using 1
-    · ring_nf at H1 ⊢
-      field_simp at H1 ⊢
-      linear_combination p * (y - z) * y ^ p * H1
-    · have H : (z / y) ^ p = z ^ p / y ^ p := div_rpow hz hy.le _
-      ring_nf at H ⊢
-      field_simp at H ⊢
-      linear_combination -H
+  · have q : 0 < y - x := by rwa [sub_pos]
+    rw [div_lt_iff q, ← div_lt_div_right hy', _root_.sub_div, div_self hy'.ne', ← div_rpow hx hy.le,
+      sub_lt_comm, ← add_sub_cancel_right (x / y) 1, add_comm, add_sub_assoc, ← div_mul_eq_mul_div,
+      mul_div_assoc, ← rpow_sub hy, sub_sub_cancel_left, rpow_neg_one, mul_assoc, ← div_eq_inv_mul,
+      sub_eq_add_neg, ← mul_neg, ← neg_div, neg_sub, _root_.sub_div, div_self hy.ne']
+    apply one_add_mul_self_lt_rpow_one_add _ _ hp
+    · rw [le_sub_iff_add_le, add_left_neg, div_nonneg_iff]; exact Or.inl ⟨hx, hy.le⟩
+    · rw [sub_ne_zero]; exact ((div_lt_one hy).mpr hxy).ne
+  · have q : 0 < z - y := by rwa [sub_pos]
+    rw [lt_div_iff q, ← div_lt_div_right hy', _root_.sub_div, div_self hy'.ne', ← div_rpow hz hy.le,
+      lt_sub_iff_add_lt', ← add_sub_cancel_right (z / y) 1, add_comm _ 1, add_sub_assoc,
+      ← div_mul_eq_mul_div, mul_div_assoc, ← rpow_sub hy, sub_sub_cancel_left, rpow_neg_one,
+      mul_assoc, ← div_eq_inv_mul, _root_.sub_div, div_self hy.ne']
+    apply one_add_mul_self_lt_rpow_one_add _ _ hp
+    · rw [le_sub_iff_add_le, add_left_neg, div_nonneg_iff]; exact Or.inl ⟨hz, hy.le⟩
+    · rw [sub_ne_zero]; exact ((one_lt_div hy).mpr hyz).ne'
 #align strict_convex_on_rpow strictConvexOn_rpow
 
-theorem convexOn_rpow {p : ℝ} (hp : 1 ≤ p) : ConvexOn ℝ (Ici 0) fun x : ℝ => x ^ p := by
+theorem convexOn_rpow {p : ℝ} (hp : 1 ≤ p) : ConvexOn ℝ (Ici 0) fun x : ℝ ↦ x ^ p := by
   rcases eq_or_lt_of_le hp with (rfl | hp)
   · simpa using convexOn_id (convex_Ici _)
   exact (strictConvexOn_rpow hp).convexOn
 #align convex_on_rpow convexOn_rpow
 
+/-- For `p : ℝ` with `0 < p < 1`, `fun x ↦ x ^ p` is strictly concave on $[0, +∞)$.
+
+We give an elementary proof rather than using the second derivative test, since this lemma is
+needed early in the analysis library. -/
```

Comment:
Is this true? I don't think this lemma's needed at all as of now (early or late). More to the point, I think proof-irrelevance applies to the reference manual: the docstring should only describe the theorem statement.